tests/integrated/test_suite

```diff
@@ -12,32 +12,87 @@ from __future__ import print_function
 import os
 import sys
 import time
+import glob
 from boututils.run_wrapper import shell
 
-# Get list of directories containing test cases
-tests=[]
-with open('test_suite_list') as testfile:
-    for test in testfile:
-        test=test.strip()
-        if test.startswith("#"):
-            pass
-        elif test.startswith("!"):
-            tests.append(test[1:])
-        else:
-            tests.append(test)
+##################################################################
+
+import argparse
+parser = argparse.ArgumentParser(description='Run or build some tests.')
+parser.add_argument("-g","--get-list",action='store_true',
+                    help='Return a list of tests that would be run/build')
+parser.add_argument("-m","--make",action='store_true',
+                    help='Build the tests, rather then run them.')
+parser.add_argument("-b","--set-bool",nargs="+",metavar=('value1=False','value2=True'),
+                    help='Set a bool value for evaluating what scripts can be run.')
+parser.add_argument("-l","--set-list",nargs="+",metavar=('test1','test2'),
+                    help='Set the tests that should be run.')
+
+args=parser.parse_args()
+
+##################################################################
+
+sys.path.append('..')
+from requirements import Requirements
+requirements = Requirements();
+
+if args.set_bool is not None:
+    lookup={'false':False,
+            'no'   :False,
+            'true' :True,
+            'yes'  :True}
+    for setbool in args.set_bool:
+        k,v=setbool.split("=")
+        v=lookup[v.lower()]
+	requirements.add(k,v,override=True)
+
+try:
+    requirements.add('make',args.make)
+except RuntimeError:
+    raise RuntimeError("The make flag needs to be set by passing --make")
 
 ##################################################################
 
+if args.set_list is None:
+    # Get list of directories containing test cases
+    try:
+        # Requires python >= 3.5
+        tests=glob.iglob('**/runtest', recursive=True)
+    except:
```

Comment:
We should catch the specific exception, which in this case is `TypeError`, I think